Don't allow use of AEAD_AES_128_CCM_8 (#2029)
quicwg/base-drafts: draft-ietf-quic-tls.md

This provides protection against off-path attackers and robustness against QUIC
version unaware middleboxes, but not against on-path attackers.

> QUIC can use any of the ciphersuites defined in {{!TLS13}} with the exception of

Reviewer: use normative language perhaps?
```suggestion
QUIC can use any of the ciphersuites defined in {{!TLS13}} but MUST NOT use
```

Reviewer 2: Yeah my understanding was that was the conclusion in #2019, but I might be incorrect.

Author: I'm happy to say that it is not specified and leave it at that.  The effect is the same.

Reviewer: @kazuho @martinthomson : I trust your judgement here, so I won't push. It's still not clear to me why disallowing is an issue. Don't mind me though.

TLS_AES_128_CCM_8_SHA256.  The AEAD for that ciphersuite, AEAD_AES_128_CCM_8
{{?CCM=RFC6655}}, does not produce a large enough authentication tag for use
with header protection ({{header-protect}}).  All other ciphersuites defined in
{{!TLS13}} have a 16-byte authentication tag and produce an output 16 bytes

> larger than their input.

Reviewer 3: It seems like there are two issues here:

1. What is sufficient to allow PNE.
2. What ciphers are secure enough.

It seems like for (1) we ought to say: "any cipher used with QUIC MUST have a minimum output size of at least X unpredictable bytes". That will exclude this cipher.

For (2) I think we ought to punt this to TLS.

Author: I broadly agree, though I don't think that we can do (1) here because we don't know what PNE scheme might be devised to be paired with some future AEAD.  It seems like we have a requirement now, but something like FFX might be OK with a smaller sample.  I would prefer, as this does, to simply say that CCM_8 doesn't have a large enough expansion to use with the header protection schemes we have defined.

Should someone address the second question (and the question of whom to convince of this is interesting, but within the IETF it's clearly TLS), and design an header protection scheme based on that, then my hope is that this text wouldn't prevent them from using that.

Reviewer 3: Hmm.... This seems like it raises the broader issue of how we import new ciphers into QUIC. As a hypothetical suppose TLS were to adopt an AES-GCM-SIV-based construction. It seems like at present we would actually require some new document to explain how to use it with QUIC. That seems sort of unfortunate, but formally I think that's true, because this document specifies a very specific list of ciphers.

If we wanted to, we could probably generalize this to some extent by documenting the *properties* of given ciphers and then how header protection works (i.e., block ciphers are done like this, nonce-based stream ciphers are done like this...) Not sure it's worth it, thoguh.

Reviewer 4: Yes, this does currently require that a PNE algorithm be defined for any new cipher to be used with QUIC v1.
